velox/row/CompactRow.cpp

```diff
@@ -184,6 +325,62 @@ int32_t CompactRow::serializeRow(vector_size_t index, char* buffer) {
   return valuesOffset;
 }
 
+void CompactRow::serializeRow(
+    vector_size_t offset,
+    vector_size_t size,
+    char* buffer,
+    const std::vector<size_t>& bufferOffsets) {
+  VELOX_CHECK_EQ(bufferOffsets.size(), size);
+
+  raw_vector<vector_size_t> rows(size);
```

Comment:
Why not use vector_size_t rows[size] here?

Reply:
What's the benefit of ` vector_size_t rows[size]`? I don't see the C-style vector used in the code base. It should be a more conventional way to use `raw_vector`

Reply:
`char inlined[StringView::kInlineSize]' is used here.
https://github.com/facebookincubator/velox/blob/main/velox/expression/CastExpr-inl.h#L588

We just allocate a fixed size array and don't need to resize it, so a simple struct is enough.
The input argument could be `vector_size_t * rows`.